SandboxDownloader-v2/Program.cs

```diff
@@ -11,89 +11,158 @@ namespace AutoCompilerForGameServer
+            var logicDurationWatch = new Stopwatch();
+
+            logicDurationWatch.Start();
+            
+            executingDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            gameServerRepository = "https://github.com/LeagueSandbox/GameServer.git";
+            repositoryBranch = "master";
+            commitMessageName = "LastCommitMessage.txt";
+            gameServerSourceFileName = "GameServer-Source";
+            copyBuildToFolder = "Compiled-GameServer";
+            needsCopied = true;
+            pauseAtEnd = true;
+            configJSON = "";
+            needsConfig = true;
+            configurationMode = "Release";
+
+            var p = new NDesk.Options.OptionSet() {
+                { "gameServerRepository=", "The game server repository",
+                    v => gameServerRepository = v
+                },
+                { "repositoryBranch=", "The game server repository branch",
+                    v => repositoryBranch = v
+                },
+                { "commitMessageName=", "Commit message file name",
+                    v => commitMessageName = v
+                },
+                { "gameServerSourceFileName=", "Game server source folder name",
+                    v => gameServerSourceFileName = v
+                },
+                { "copyBuildToFolder=", "The folder that the build gets copied to",
+                    v => copyBuildToFolder = v
+                },
+                { "needsCopied=", "Does it need copied even if it doesn't need built?",
+                    (bool v) => needsCopied = v
+                },
+                { "needsConfig=", "Does it need JSON config?",
+                    (bool v) => needsConfig = v
+                },
+                { "pauseAtEnd=", "Should it be pasued at the end?",
+                    (bool v) => pauseAtEnd = v
+                },
+                { "configJSON=", "The config JSON for the compiled game server.",
+                    v => configJSON = v
+                }
+            };
+            
+            try
+            {
+                p.Parse(args);
+            }
+            catch (NDesk.Options.OptionException e)
+            {
+                Console.Write("Command line error: ");
+                Console.WriteLine(e.Message);
+                return;
+            }
+
             Console.WriteLine("Welcome to my GameServer updater!");
             var lastCommit = GetLastCommitMessageFromWeb();
-            if (!File.Exists(Path.Combine(ExecutingDirectory, "LastCommitMessage.txt")))
+
+            var needsCompiled = false;
+
+            if (!File.Exists(Path.Combine(executingDirectory, commitMessageName)))
             { // First run
-                DownloadAndCompileServer();
-                File.WriteAllText(Path.Combine(ExecutingDirectory, "LastCommitMessage.txt"), lastCommit);
+                File.WriteAllText(Path.Combine(executingDirectory, commitMessageName), lastCommit);
```

Comment:
Updating that file should be last, since you cannot get if it downloaded and compiled well or user shutdown the program mid-download